test/e2e/prometheus_test.go

```diff
@@ -3730,13 +3730,22 @@ func testPromMinReadySeconds(t *testing.T) {
 		t.Fatal("Creating prometheus failed: ", err)
 	}
 
-	promSS, err := kubeClient.AppsV1().StatefulSets(ns).Get(context.Background(), "prometheus-basic-prometheus", metav1.GetOptions{})
-	if err != nil {
-		t.Fatal(err)
-	}
+	var got int32
+	err = wait.Poll(time.Second, time.Minute*5, func() (bool, error) {
+		promSS, err := kubeClient.AppsV1().StatefulSets(ns).Get(context.Background(), "prometheus-basic-prometheus", metav1.GetOptions{})
+		if err != nil {
+			return false, err
```

Comment:
I think this that returning an error will short-circuit the wait.Poll function and won't really reduce the flake. What we want is 
```
if err != nil {
  return false, nil
}
```

Reply:
Hey @fpetkovski - I'm not entirely sure I agree since from https://github.com/prometheus-operator/prometheus-operator/pull/4406/files#diff-b3d3450b51af206bb0009c2b9c2f863c3d03a3575d64cbdb18510a2bc834144aR3728 we expect the statefulset to exist which is why I kept the hard error. I don't *think* that is what was causing the flake, it was because the spec hadn't yet been updated which the retry logic should solve.

Our other tests don't retry on error for the `Get Resource` calls so I would not expect this to behave or be handled any differently.

That said, I don't think changing it to retry on error is a big issue so I can do it if you prefer. Just telling you my thought process :)

Reply:
I see, I was under the impression that the flake was caused because we didn't wait long enough. In that case, it's correct to do what you've done and return an error if the sts is missing :+1: